makefiles/utils/variables.mk

```diff
@@ -0,0 +1,33 @@
+# Utilities to set variables and environment for targets
+# These functions should help replacing immediate evaluation and global 'export'
+
+
+# Evaluate a deferred variable only on once on its first usage
```

Comment:
typo: double `on`, remove before `once`

Reply:
I can see that I piggybacked that change from a wip branch a bit too fast :D